File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        mymap = {}
        for i in range(len(nums)):
            
            subtraction = target - nums[i]
            if(subtraction in mymap):
                return [mymap[subtraction], i]
            mymap[nums[i]] = i

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map1;
        vector<int> ans;
        
        for(int i=0; i<=nums.size(); i++){
            int subtraction = target - nums[i];
            
            //agar mil gaya to 
            if(map1.find(subtraction) != map1.end()){ 
                return {map1[subtraction], i};
            
             }       
            map1[nums[i]] = i;
        
        }
        return ans;
        
    }
};